sovereign-forge/src/phases.rs

```diff
@@ -0,0 +1,92 @@
+use core::ops::Deref;
+use proxies::chain_factory_proxy::ChainFactoryContractProxy;
+use transaction::StakeMultiArg;
+
+use multiversx_sc::{
+    require,
+    types::{ManagedVec, MultiValueEncoded, ReturnsResult},
+};
+
+use crate::{
+    common::{
+        self,
+        utils::{ChainContractsMap, ContractInfo, ScArray},
+    },
+    err_msg,
+};
+
+#[multiversx_sc::module]
+pub trait PhasesModule: common::utils::UtilsModule + common::storage::StorageModule {
+    #[payable("EGLD")]
+    #[endpoint(deployPhaseOne)]
+    fn deploy_phase_one(
+        &self,
+        min_validators: u64,
+        max_validators: u64,
+        min_stake: BigUint,
+        additional_stake_required: MultiValueEncoded<StakeMultiArg<Self::Api>>,
+    ) {
+        let call_value = self.call_value().egld_value();
+        self.require_correct_deploy_cost(call_value.deref());
+
+        let blockchain_api = self.blockchain();
+        let caller = blockchain_api.get_caller();
+        let caller_shard_id = blockchain_api.get_shard_of_address(&caller);
+
+        let chain_id = self.generate_chain_id();
+
+        let chain_factories_mapper = self.chain_factories(caller_shard_id);
+        require!(
+            !chain_factories_mapper.is_empty(),
+            "There is no Chain-Factory address registered in the {} chain",
+            chain_id
```

Comment:
Should be like this. I don't know if shard id of type u32 can be printed there
```suggestion
            "There is no Chain-Factory address registered in shard {}",
            caller_shard_id
```